src/components/application_manager/test/application_manager_impl_test.cc

```diff
@@ -35,78 +35,236 @@
+        .WillByDefault(Return(true));
+    ON_CALL(mock_application_manager_settings_, stop_streaming_timeout())
+        .WillByDefault(Return(1u));
+    ON_CALL(mock_application_manager_settings_, default_timeout())
+        .WillByDefault(ReturnRef(timeout));
+    app_manager_impl_.reset(new am::ApplicationManagerImpl(
         mock_application_manager_settings_, mock_policy_settings_));
 
-    app_id_ = 10;
+    ASSERT_TRUE(app_manager_impl_.get());
   }
 
   uint32_t app_id_;
-  MockApplicationManagerSettings mock_application_manager_settings_;
-  policy_test::MockPolicySettings mock_policy_settings_;
+  NiceMock<policy_test::MockPolicySettings> mock_policy_settings_;
+  utils::SharedPtr<NiceMock<resumption_test::MockResumptionData>> mock_storage;
+  NiceMock<con_test::MockConnectionHandler> mock_connection_handler;
+  NiceMock<protocol_handler_test::MockSessionObserver> mock_session_observer;
+  NiceMock<hmi_message_handler_test::MockHMIMessageHandler>
+      mock_hmi_mess_handler;
+  NiceMock<MockApplicationManagerSettings> mock_application_manager_settings_;
+  std::auto_ptr<am::ApplicationManagerImpl> app_manager_impl_;
+};
 
-  am::ApplicationManagerImpl& get_app_manager() {
-    return (*app_manager_impl.get());
+struct AppIdExtractor {
+  AppIdExtractor() {
+    hmi_app_id_ = 5u;
+    is_called_ = false;
   }
 
- private:
-  std::auto_ptr<am::ApplicationManagerImpl> app_manager_impl;
+  bool operator()(
+      const hmi_message_handler::MessageSharedPointer& message) const {
+    using namespace application_manager::strings;
+    sync_primitives::AutoLock auto_lock(state_lock_);
+    bool retVal = false;
+    Json::Reader reader;
+    Json::Value json_value;
+    const std::string str = message->json_message();
+    if (reader.parse(str, json_value, false)) {
+      hmi_app_id_ = json_value[params][applications][0][app_id].asUInt();
+      is_called_ = true;
+      retVal = true;
+    }
+    state_condition_.Broadcast();
+    return retVal;
+  }
 };
 
 TEST_F(ApplicationManagerImplTest,
+       FindAppToRegister_SearchValidHmiId_ExpectSuccess) {
+  using namespace NsSmartDeviceLink::NsSmartObjects;
+  SmartObject app_data;
+  const uint32_t connection_key = 65537u;
+  const uint32_t app_id = 8975689u;
+  app_data[am::json::name] = "application_manager_test";
+  app_data[am::json::appId] = app_id;
+  app_data[am::json::android] = "bucket";
+  app_data[am::json::android][am::json::packageName] = "com.android.test";
+  smart_objects::SmartObject sm_object(SmartType_Map);
+  sm_object[am::json::response][0] = app_data;
+
+  EXPECT_CALL(mock_hmi_mess_handler, SendMessageToHMI(Truly(AppIdExtractor())))
+      .Times(AtLeast(1u));
+
+  app_manager_impl_->ProcessQueryApp(sm_object, connection_key);
+
+  {
+    sync_primitives::AutoLock auto_lock(state_lock_);
+    while (!is_called_) {
+      ASSERT_TRUE(sync_primitives::ConditionalVariable::kTimeout ==
+                  state_condition_.Wait(auto_lock));
+    }
```

Comment:
@anosach-luxoft,  while here need to be ensure that we are free by right event. But you right, here mistake I forgot `WaitFor` and assert will work when timeout happens. I'll fix in next commit.